File: src/main/node/common/dummy-ai.ts
export class DummyAI {
    static async getCompanyAttractiveness(companySymbol: string): Promise<{ id: string; score: number }> {
        return new Promise((r) => {
            setTimeout(() => {
                r({
                    id: companySymbol,
                    score: Math.round(Math.random() * 100) / 100,
                });
            }, 1000);
        });
    }
}

File: src/main/python/common/dummy_ai.py
import random
from time import sleep
from typing import TypedDict

IScoredSymbol = TypedDict(
    "IScoredSymbol",
    {
        "id": str,
        "score": float,
    },
)


def getCompanyAttractiveness(companySymbol: str) -> IScoredSymbol:
    """Calculate company attractiveness using a complex algorithm

    Args:
        companySymbol (str): Company symbol

    Returns:
        IScoredSymbol: Dict object containing the company symbol and its score
    """
    sleep(1)
    return {"id": companySymbol, "score": round(random.uniform(0, 1), 2)}
